### ISSUE TITLE
Crashbug with new from template

### ISSUE THREAD

**Issue by ReinerBforartists** [Bforartists/Bforartists#1431]
Create a new scene with new from template, vfx. Create again a new scene with new from template,  vfx. Instead of creating a new blank vfx scene Bforartists closes down. Happens with sculpting too.

The Blender layout is different here. They start with tracking in the vfx layout. And Blender does not close when creating the vfx layout again.

![crash](https://user-images.githubusercontent.com/13919702/77854373-40b34d80-71ea-11ea-8460-400552ed8480.jpg)

**Comment by ReinerBforartists** [comment]
Unfunny one. I thought it's the startup.blends. But even with a Blender startup.blend it crashes. And even unfunnier. the vital parts are equal in Blender. This one will make me lots of fun for sure ...

![crash](https://user-images.githubusercontent.com/13919702/77885841-0e483580-7268-11ea-8e44-58b27fd4de02.jpg)

**Comment by ReinerBforartists** [comment]
Hm, the all template does not contain all tabs. And is not crashing ...

**Comment by ReinerBforartists** [comment]
Introduced between bfa 0.8.0 and 0.9.0. And i can't even compile this old version since the libs has changed meanwhile -.-

**Comment by ReinerBforartists** [comment]
Still stuck here. Renaming the temlates in question fixes the crashes. But this is no good workaround. We can't call our templates different than the ones in Blender.

![renamed](https://user-images.githubusercontent.com/13919702/78216955-48f5dc00-74bb-11ea-9369-6b55495bb5b9.jpg)

**Comment by ReinerBforartists** [comment]
The All template is the one from us. The other templates are the Blender templates. And this would explain wy All works, and why renaming the templates works. 

But the files in question, like versioning_defaults.c, are still equal -.-

**Comment by Draise14** [comment]
## -------------- Test 1
I gave a quick test from 0.9.1 installation: 

- **Motion Tracking** and **Masking** from VFX and All is ok - no crash. 
- The **Compositing** and **Rendering** workspaces crashes from the VFX and All section specifically. 
- The **Rendering** crashes from all sections generally. 

Conclusion, there are two corrupt workspaces with my setup, so when you try add a new workspace or generate a new group of workspaces with that particular corrupt space, it will crash. This kind of crash I've had before when I had an old version of a workspace migrated over from an former version, and they weren't updated. I had to practically rebuild and re-save the startup.blend files to fix. 

To test again just to make sure:

1. I made a new startup from the default with only the two workspaces I could load from VFX (Masking and Motion Tracking) 
2. Overwrote the VFX startup blend. 
3. Loaded the VFX layout, no problem, no crashes. (I had the default Startup Blend backed up in the process). 
4. Conclusion, the VFX layout without "rendering" and "composite" removes the crash. Those workspaces must be corrupt. 

_My observations might be biased to my own system as I migrated work where the startup.blend files were made in 0.9.0 or even before hand.. I probably should update those..._ 

## -------------- Test 2
Ok, did a factory reset and backed-up my config folder. 
When I did the factory reset, I noticed no startup.blend files were written nor saved, even after first use and userpref save. 
Now, when making new workspace groups from VFX, then yes, I can reproduce the crash. 
So.. loading up a default factory setup will crash eventually as it's pulling the workspace startup.blend from.. who knows where? 

Now here is the mystery:
When I save the startup.blend groups individually so they now are in the config folder, then load, and load and load again... it then pulls them from the config folder even as defaults, and there is no issue there ("fixed"). When the VFX startup.blend file is saved in the config folder, it won't crash. 

To reproduce a "fix":
- Load BFA first time, save userprefs to create the config folder.
- Take note that the Config folder doesn't have the workspaces listed
- Load VFX template multiple times, crash. 
- Load BFA again. 
- Load the VFX template on first use, once. 
- Save default startup.blend for the VFX group
- Take note that it's now saved to the Config folder
- Create new workspaces multiple times, no crash. 

Observations... I guess a factory reset and load from factory can only be done once, then when you try do it again it might be assuming that the config startup.blend setup is saved and will look there second time.. giving you an Access Violation - because nothing is there. Until something is there, the bug goes away.

**Comment by ReinerBforartists** [comment]
Interesting observation, thanks for that. But Blender does not crash at all. And it has also no data written to the appdata folder. So this is unfortunately not of big help.

That you first need to click at the next button in the splash screen to save a first user configuration to make changes at the startup.blend working is a known Blender bug. Conmfirmed, marked as low priority, and now happily forgotten for the next 20 years.

**Comment by Draise14** [comment]
... How does blender know when to read from the installation folder then to the %appdata% only after saving userpref's?  Maybe that is where it's broken: BFA is looking in the %appdata% from the start, and that is why we get an Access Violation...

**Comment by ReinerBforartists** [comment]
I just know that the code is the same from what i can see.

**Comment by ReinerBforartists** [comment]
Also here, Successfully reproduced at throwing in the startup.blend files and by copying over the userdef_default.c file from Bforartists to Blender. Now let's have a closer look which one of those is the baddie ...

**Comment by ReinerBforartists** [comment]
It's the startup.blend

Unfunny part: using Bforartists startup.blend in Blender makes Blender crash the same. Using Blender startup.blend in Bforartists still crashes ...

**Comment by ReinerBforartists** [comment]
Ah, wrong 50%, it's the userdef_default. Using this one in Bforartists makes the crash go away. Analyzing ...

**Comment by ReinerBforartists** [comment]
Has to do with Load UI off. I was able to reproduce this crash in Blender.

**Comment by ReinerBforartists** [comment]
Blender bugreport created https://developer.blender.org/T76392

**Comment by ReinerBforartists** [comment]
Fixed with the latest merge.

**Comment by floorheight** [comment]
I'm having consistent crashes on the 3.2.0 AppImage and tar versions in Manjaro Linux (doesn't use deb so can't try that). The 3.1.3 version works fine with all "New from" templates able to load. BFA 3.2.0 crashes every time when trying to load the "Sculpting" template. Both the splash-screen and menu options trigger this behaviour.

**Comment by Draise14** [comment]
Can confirm @floorheight, I will make a new bug report.
